scipy/ndimage/_interpolation.py

```diff
@@ -841,6 +842,9 @@ def zoom(input, zoom, output=None, order=3, mode='constant', cval=0.0,
     complex_output = np.iscomplexobj(input)
     output = _ni_support._get_output(output, input, shape=output_shape,
                                      complex_output=complex_output)
+    if all(z == 1 for z in zoom) and prefilter:  # early exit for gh-20999
```

Comment:
Sorry if this is mentioned elsewhere but it's not entirely obvious to me why the early exit only applies when `prefilter=True`?

Reply:
I was thinking that If `prefilter=False`, the argument has been prefiltered to get spline coefficients, so `input` is no longer really an image. it needs to undergo additional processing to return an image.

```python3
import matplotlib.pyplot as plt
from scipy import datasets, ndimage
img = datasets.face()
filtered = ndimage.spline_filter(img)
plt.imshow(filtered)  # this is what it would look like with early exit

# out = np.zeros_like(img)
# out[:] = ndimage.zoom(filtered, 1, prefilter=False)
# plt.imshow(out)  # this is what we expect
```

Reply:
The docs say

> If setting this to False, the output will be slightly blurred if order > 1, unless the input is prefiltered

so the early exit would avoid the doumented blurring for non pre-filtered input.

That said, maybe the `zoom=1` argument trumps that and should still be a no-op, I don't know.

Reply:
`zoom=1` does not mean no-op on the `input`; it means return an image identical to the original image. If `prefilter` is False, `input` is not the original image, so processing is needed to return the original image.

Reply:
yes, that sounds right to me

Reply:
Ah I see, would you be able to a quick comment to avoid any future confusion

Reply:
```suggestion
    if all(z == 1 for z in zoom) and prefilter:  # early exit for gh-20999
        # zoom 1 means "return original image". If `prefilter=False`,
        # `input` is *not* the original image; processing is still needed
        # to undo the filter. So we only early exit if `prefilter`.
```
If this looks safe, please feel free to commit with skip CI and merge.